flink-yarn-tests/src/test/java/org/apache/flink/yarn/YARNSessionFIFOITCase.java

```diff
@@ -196,15 +197,16 @@ void runDetachedModeTest(Map<String, String> securityProperties) throws Exceptio
             ApplicationId id = app.getApplicationId();
             yc.killApplication(id);
 
-            while (getApplicationReportWithRetryOnNPE(yc, EnumSet.of(YarnApplicationState.KILLED))
-                                    .size()
-                            == 0
-                    && getApplicationReportWithRetryOnNPE(
-                                            yc, EnumSet.of(YarnApplicationState.FINISHED))
-                                    .size()
-                            == 0) {
-                sleep(500);
-            }
+            CommonTestUtils.waitUntilCondition(
+                    () ->
+                            !getApplicationReportWithRetryOnNPE(
+                                            yc,
+                                            EnumSet.of(
+                                                    YarnApplicationState.KILLED,
+                                                    YarnApplicationState.FINISHED))
+                                    .isEmpty(),
+                    Deadline.fromNow(Duration.ofSeconds(timeoutInSecs)),
```

Comment:
same as above; since there was no previous timeout, let's be conservative and give it a few minutes.